src/styles/main.scss

```diff
@@ -1,3 +1,65 @@
+$cell-size: 100px;
+$col-count: 7;
+$gap-size: 1px;
+$padding-size: 10px;
+
 body {
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  width: $col-count * ($cell-size + $gap-size);
+  padding: $padding-size;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    box-sizing: border-box;
+    background-color: #eee;
+    height: $cell-size;
+    width: $cell-size;
+    border: 1px solid black;
+  }
+
+  @for $number from 1 through 31 {
+    &__day:nth-child(#{$number})::before {
+      content: "#{$number}";
+      text-align: center;
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+      display: block;
+    }
+  }
+
+  &__day:hover {
+    cursor: pointer;
+    background-color: pink;
+    transform: translateY(-20px);
+    transition-duration: 0.5s;
+  }
+
+  $weekdays:
+    "mon" 0,
+    "tue" 1,
+    "wed" 2,
+    "thu" 3,
+    "fri" 4,
+    "sat" 5,
+    "sun" 6;
+
+  @each $day, $number in $weekdays {
+    &--start-day--#{$day} > .calendar__day:first-child {
+      margin-left: $number * ($cell-size + $gap-size);
+    }
+  }
+
+  @for $length from 28 through 30 {
```

Comment:
The @for loop should include the 31 value. Change to '@for $length from 28 through 31'.